File: 프로그래머스/unrated/181890. 왼쪽 오른쪽/왼쪽 오른쪽.java
class Solution {
    public String[] solution(String[] str_list) {
        int len = str_list.length;
        String[]ans = new String[0];
        for (int i = 0; i < len; i++)
        {
            if (str_list[i].equals("l"))
            {
                ans = new String[i];
                System.arraycopy(str_list, 0, ans, 0, i);
                break;
            }
            if (str_list[i].equals("r"))
            {
                ans = new String[len - i - 1];
                System.arraycopy(str_list, i + 1, ans, 0, len - i - 1);
                break;
            }
        }
        return ans;
    }
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.js
function solution(str_list) {
  const i = str_list.findIndex((str) => /l|r/.test(str));
  return (i === -1 ? [] : (str_list[i] === 'l' ? str_list.slice(0, i) : str_list.slice(i + 1)));
}